mapr401/src/org/pentaho/hadoop/shim/mapr401/ConfigurationProxyV2.java

```diff
@@ -38,16 +39,39 @@
 
   private Job job;
 
+  protected class JobProxy extends Job {
+    private JobProxy( JobConf conf ) throws IOException {
+      super( conf );
+    }
+
+    void refreshUGI() {
+      try {
+        this.ugi = UserGroupInformation.getCurrentUser();
+      } catch ( IOException e ) {
+        throw new RuntimeException( e );
+      }
+    }
+  }
+
   public ConfigurationProxyV2() throws IOException {
-    job = Job.getInstance();
-    getJobConf().addResource( "hdfs-site.xml" );
+    // create with a null Cluster
+    JobConf jobConf = new JobConf( new org.apache.hadoop.conf.Configuration() );
+    job = new JobProxy( jobConf );
+    job.getConfiguration().addResource( "hdfs-site.xml" );
+    job.getConfiguration().addResource( "hbase-site.xml" );
+    job.getConfiguration().addResource( "core-site.xml" );
+    job.getConfiguration().addResource( "yarn-site.xml" );
+    job.getConfiguration().addResource( "mapred-site.xml" );
+    job.getConfiguration().addResource( "oozie-site.xml" );
+    job.getConfiguration().addResource( "hive-site.xml" );
```

Comment:
Aren't some of these (like hdfs-site.xml) added automatically in the base Configuration?